include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -829,6 +829,21 @@ namespace internal
       }
 #endif
 
+#ifdef DEAL_II_TRILINOS_WITH_TPETRA
+      template <typename Number>
+      void
+      copy_locally_owned_data_from(
+        const LinearAlgebra::TpetraWrappers::Vector<Number> &src,
+        LinearAlgebra::distributed::Vector<Number>          &dst)
+      {
+        // ReadWriteVector does not work for ghosted
+        // TrilinosWrappers::MPI::Vector objects. Fall back to copy the
+        // entries manually.
+        for (const auto i : dst.locally_owned_elements())
+          dst[i] = src[i];
+      }
+#endif
```

Comment:
If we merge this patch with this part as-is, can you open a github issue about this? We would really want to make `ReadWriteVector` work for this operation.

Out of curiosity, what is it that does not work?

Reply:
If we do not find the problem in this PR, I will open an Issue.